pkg/http/round_tripper_with_headers.go

```diff
@@ -58,3 +60,53 @@ func (r *RoundTripperWithHeaders) RoundTrip(req *http.Request) (*http.Response,
 
 	return r.next.RoundTrip(req)
 }
+
+// Client is a wrapper around the Go stdlib http client.
+type Client struct {
+	internal *http.Client
+}
+
+// NewClient returns a new Client with its internal http client
+// set to the default client.
+func NewClient() *Client {
+	return &Client{
+		internal: http.DefaultClient,
+	}
+}
+
+type GetOptions func(*http.Request)
+
+func WithHeader(key, value string) GetOptions {
+	return func(r *http.Request) {
+		r.Header.Add(key, value)
+	}
+}
+
+func WithJSONAccept() GetOptions {
+	return func(r *http.Request) {
+		r.Header.Add("Accept", "application/json")
+	}
+}
+
+// GetWithContext performs a Get request with the context provided.
+func (c *Client) GetWithContext(ctx context.Context, url string, opts ...GetOptions) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, opt := range opts {
+		opt(req)
+	}
+
+	resp, err := c.internal.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("http get: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New(resp.Status)
+	}
+
+	return resp, nil
+}
```

Comment:
Why do we need this additional client? We already have a separate client in base/api/handlers/requesthandler?

Reply:
can this functionality just be added there?

Reply:
this is an internal general http client and doesnt need all of the requestHandler functionality.  It makes sense to me to only have the Fetcher use a client that gives it the functionality that it needs